src/events.rs

```diff
@@ -0,0 +1,61 @@
+use crate::context::{ConcreteBaseAudioContext};
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum EventType {
+    Ended
+}
+
+// maybe not the right, but needs to review according to type EventHandler in capacity.rs
+pub(crate) struct EventHandlerInfos {
+    // could be Option w/ None meaning that its a context event (cf. onSinkChange, onStateChange, etc.)
+    pub node_id: u64,
+    pub event_type: EventType,
+    pub callback: Box<&'static (dyn Fn() + Send + Sync + 'static)>,
+}
+
+// for control thread entities
+pub(crate) trait EventListener {
```

Comment:
I don't think this trait needs to exist. It can just be a method on the ConcreteBase